test/integration/emitter_test.cpp

```diff
@@ -1212,6 +1212,83 @@ TEST_F(EmitterTest, NaN) {
 	  "bar: .nan");
 }
 
+TEST_F(EmitterTest, ComplexFlowSeqEmbeddingAMapWithNewLine) { 
+  out << YAML::BeginMap;
+
+  out << YAML::Key << "NodeA" << YAML::Value << YAML::BeginMap;
+  out << YAML::Key << "k" << YAML::Value << YAML::Flow << YAML::BeginSeq;
+  out << YAML::BeginMap << YAML::Key << "i" << YAML::Value << 0 << YAML::EndMap
+      << YAML::Newline;
+  out << YAML::BeginMap << YAML::Key << "i" << YAML::Value << 1 << YAML::EndMap
+      << YAML::Newline;
+  out << YAML::EndSeq;
+  out << YAML::EndMap;
+
+  out << YAML::Key << "NodeB" << YAML::Value << YAML::BeginMap;
+  out << YAML::Key << "k" << YAML::Value << YAML::Flow << YAML::BeginSeq;
+  out << YAML::BeginMap << YAML::Key << "i" << YAML::Value << 0 << YAML::EndMap
+      << YAML::Newline;
+  out << YAML::BeginMap << YAML::Key << "i" << YAML::Value << 1 << YAML::EndMap
+      << YAML::Newline;
+  out << YAML::EndSeq;
+  out << YAML::EndMap;
+
+  out << YAML::EndMap;
+
+  ExpectEmit(R"(NodeA:
+  k: [{i: 0},
+    {i: 1},
+    ]
+NodeB:
+  k: [{i: 0},
+    {i: 1},
+    ])");
+}
+
+TEST_F(EmitterTest, ComplexFlowSeqEmbeddingAMapWithNewLineUsingAliases) {
+  out << BeginMap;
+
+  out << Key << "Node" << Anchor("Node") << Value << BeginMap;
+  out << Key << "k" << Value << Flow << BeginSeq;
+  out << BeginMap << Key << "i" << Value << 0 << EndMap;
+  out << YAML::Newline;
+  out << BeginMap << Key << "i" << Value << 1 << EndMap;
+  out << YAML::Newline;
+  out << EndSeq << EndMap;
+
+  out << Key << "NodeA" << Alias("Node");
+  out << Key << "NodeB" << Alias("Node");
+
+  out << EndMap;
+
+  ExpectEmit(R"(Node: &Node
+  k: [{i: 0},
```

Comment:
Can put an anchor on each of these nodes (the sequence, each map in the sequence, and both key and value within each map)?

(And when you do tags, please do the same for that.)

Reply:
This kind of yaml file not sure is possible to emit:

```yml
Aliases:
 - {&i i: &v0 0}
 - {&i i: &v1 1}
NodeA:
  k: [{*i : *v0}, {*i : *v1}]
NodeB:
  k: [*i : *v0, *i : *v1]
```

should be a valid yaml, using to validate: https://yamlvalidator.com/


the actual test that is failing will be:
```c
TEST_F(EmitterTest, ComplexFlowSeqEmbeddingAMapUsingAliases4) {
  out << BeginMap;

  out << Key << "Aliases" << Value << BeginSeq;
  out << BeginMap << Anchor("i") << Key << "i" << Anchor("v0") << Value << 0
      << EndMap;
  out << BeginMap << Key << Anchor("i") << "i" << Value << Anchor("v1") << 1
      << EndMap;
  out << EndSeq;

  out << Key << "NodeA" << Value << BeginMap;
  out << Key << "k" << Value << Flow << BeginSeq;
  out << BeginMap << Key << Alias("i") << Value << Alias("v0") << EndMap;
  out << BeginMap << Key << Alias("i") << Value << Alias("v1") << EndMap;
  out << EndSeq << EndMap;

  out << Key << "NodeB" << Value << BeginMap;
  out << Key << "k" << Value << Flow << BeginSeq;
  out << Key << Alias("i") << Value << Alias("v0");
  out << Key << Alias("i") << Value << Alias("v1");
  out << EndSeq << EndMap;

  out << EndMap;

  ExpectEmit(R"(Aliases:
 - {&i i: &v0 0}
 - {&i i: &v1 1}
NodeA:
  k: [{*i : *v0}, {*i : *v1}]
NodeB:
  k: [*i : *v0, *i : *v1])");
}
```

This test is going to fail with the actual master code, so should be considered in a different issue/PR to avoid scope creep in this one. Of course there analog test checking the newline will fail too.


Let me know your thoughts, i can commit anyway including this pair of tests and the CI should fail too.


p.s.
this test is the one asked for ` both key and value within each map`
[maybe consider a better naming for the review too, at this moment i am not in a really poetic mind form :)]

Reply:
Thanks for investigating. See below; if you remove the failing test, this PR looks good.